### DISCUSSION TITLE
How to disable fingerprint of static assets in .NET 9?

### DISCUSSION THREAD

**Discussion by Tornhoof** [dotnet/aspnetcore#58910] [Q&A] [answered]
Is it possible to disable the fingerprintg of static web assets?

Motivation:
For some unknown reason I never figured fully out, one of my web projects using a nuget with a blazor app in it, never worked properly with the automatic way to link the stylesheet of the blazor app.

So ``<link rel="stylesheet" href="@Assets["BlazorApp1.styles.css"]" /> `` in the App.razor never worked properly, instead I linked the BlazorApp1.bundle.scp.css directly in my app.css. But as the name now contains a fingerprint this gets rather complicated.

**Comment by ViRuSTriNiTy** [comment, marked as answer]
You could try to set the MSBuild property [`StaticWebAssetFingerprintingEnabled`](https://github.com/dotnet/sdk/blob/2d87cb8327aa0d998bc5a5e34cbcf6585c8dc5af/src/StaticWebAssetsSdk/Targets/Microsoft.NET.Sdk.StaticWebAssets.targets#L433) to `false` like

```
<PropertyGroup>
  <StaticWebAssetFingerprintingEnabled>false</StaticWebAssetFingerprintingEnabled>
</PropertyGroup>
```

**Comment by Tornhoof** [comment]
Thank you very much, this indeed works. Looking for that property does not match anything in this repo, after searching more widely I found the property in the sdk repo.

**Reply by ViRuSTriNiTy** [reply]
> Thank you very much, this indeed works. Looking for that property does not match anything in this repo, after searching more widely I found the property in the sdk repo.

Yep, I forgot to add a link to the property definition while writing my answer. Fixed 😎

**Comment by VahidN** [comment]
The mentioned setting doesn't disable creating `obj\Debug\net9.0\compressed` folder.

**Reply by wisamidris77** [reply]
Yes but you can disable it using DisableBuildCompression property

```xml
<PropertyGroup>
  <DisableBuildCompression>true</DisableBuildCompression>
</PropertyGroup>
```
